buildSrc/src/main/kotlin/com/virtuslab/gitmachete/buildsrc/BuildUtils.kt

```diff
@@ -0,0 +1,181 @@
+
+            kotlinOptions.jvmTarget = javaMajorVersion.majorVersion
+        }
+    }
+
+    fun addIntellijToCompileClasspath(project: Project, params: Map<String, Boolean>) {
+        val tasksBefore = mutableListOf<Task>()
+        tasksBefore.addAll(project.tasks)
+
+        project.apply<IntelliJPlugin>()
+
+        val tasksAfter = mutableListOf<Task>()
+        tasksAfter.addAll(project.tasks)
+        tasksAfter.removeAll(tasksBefore)
+        // For the frontend subprojects we only use gradle-intellij-plugin to provide dependencies,
+        // but don't want the associated tasks to be available; they should only be available in the root project.
+        tasksAfter.forEach { it.enabled = false }
+        // The only task (as of gradle-intellij-plugin v1.7.0, at least) that needs to be enabled in all IntelliJ-aware modules
+        // is `classpathIndexCleanup`, to avoid caching issues caused by `classpath.index` file
+        // showing up in build/classes/ and build/resources/ directories.
+        // See https://github.com/JetBrains/gradle-intellij-plugin/issues/1039 for details.
+        project.tasks.withType<ClasspathIndexCleanupTask> {
+            enabled = true
+        }
+
+        project.configure<CheckerFrameworkExtension> {
+            // Technically, UI thread handling errors can happen outside of the (mostly frontend) modules that depend on IntelliJ,
+            // but the risk is minuscule and not worth the extra computational burden in every single build.
+            // This might change, however, if/when Checker Framework adds @Heavyweight annotation
+            // (https://github.com/typetools/checker-framework/issues/3253).
+            checkers.add("org.checkerframework.checker.guieffect.GuiEffectChecker")
+        }
+
+        project.configure<IntelliJPluginExtension> { // or configure IntellijPluginExtension?
+            version.set(IntellijVersionHelper.getInstance()["buildTarget"] as String)
+            // No need to instrument Java classes with nullability assertions, we've got this covered much better by Checker
+            // (and we don't plan to expose any part of the plugin as an API for other plugins).
+            instrumentCode.set(false)
+            if (params["withGit4Idea"] == true) {
+                plugins.set(listOf("git4idea"))
+            }
+        }
+    }
+
+    fun commonsIO(project: Project) {
+        project.dependencies {
+            val dependencyNotation = "commons-io:commons-io:2.11.0"
+            add("implementation", dependencyNotation)
+        }
+    }
+
+    fun ideProbe(project: Project) {
+        project.repositories {
+            // Needed for com.intellij.remoterobot:remote-robot
+            maven { url = `java.net`.URI("https://packages.jetbrains.team/maven/p/ij/intellij-dependencies") }
+        }
+
+        project.dependencies {
+            add("uiTestImplementation", testFixtures(project(":testCommon")))
```

Comment:
Is it possible to do sth like
```suggestion
            uiTestImplementation(testFixtures(project(":testCommon")))
```
?

Reply:
uiTestImplementation is not recognized outside of build scripts, probably it's a problem with the [ExtensionAware](https://docs.gradle.org/current/dsl/org.gradle.api.plugins.ExtensionAware.html) interface.